### PR TITLE
Generate the source code reference

### PR DESCRIPTION
**Pull request by m-vdb**
## Proposed changes
- Add a `docusaurus` renderer here: https://github.com/RasaHQ/pydoc-markdown/pull/1
- This renderer autogenerates the docs + a sidebar. We can include this sidebar in our sidebar conf and magic happens
Screenshot 2020-08-05 at 15 50 40

## Checklist
- [ ] some docs will need revamping, either do it on this PR or in a separate PR with CSE

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -58,6 +58,8 @@ install-full-windows: install install-mitie
 	poetry install -E spacy -E transformers -E jieba
 
 install-docs:
+	# FIXME: poetry doesn't support installing from a git subdirectory
+	poetry run pip install git+https://github.com/RasaHQ/pydoc-markdown.git@23d825f#subdirectory=pydoc-markdown
```
Comment (m-vdb) [inline Makefile:62]: really not idea but not sure there is a workaround

File: docs/sidebars.js [hunk 2]
```diff
@@ -103,7 +103,10 @@ module.exports = {
       type: 'category',
       label: 'Reference',
       collapsed: false,
-      items: ['glossary'],
+      items: [
+        'glossary',
+        require("./docs/reference/sidebar.json"),
```
Comment (m-vdb) [inline docs/sidebars.js:108]: this is generated by the script

File: Makefile (hunk 1 above)
Comment (tmbo) [reply Makefile:62]: can we just publish and install from our pypi.rasa.com?

Comment (m-vdb) [reply Makefile:62]: happy to do it; can you share credentials in 1password?

**Comment by tmbo** [comment]
merged since it already passed tests

**Pull request merged**